src/components/Item/Item.jsx

```diff
@@ -74,37 +78,246 @@ class Item extends Component {
     let uri = this.props.match.url;
     let params = this.props.match.params;
     let hypertopic = new Hypertopic(conf.services);
-    hypertopic.getView(uri).then((data) => {
+    hypertopic.getView(uri).then(data => {
       let item = data[params.corpus][params.item];
-      item.topic = (item.topic) ? groupBy(item.topic, ['viewpoint']) : [];
+      item.topic = item.topic ? groupBy(item.topic, ['viewpoint']) : [];
       this.setState(item);
     });
   }
+
+  _assignTopic(topicToAssign, viewpointId) {
+    let hypertopic = new Hypertopic(conf.services);
+    const ressourceId = this.state.resource[0].substr(
+      this.state.resource[0].lastIndexOf('/') + 1
+    );
+
+    return hypertopic
+      .get({ _id: ressourceId })
+      .then(data => {
+        data.topics[topicToAssign.id] = { viewpoint: viewpointId };
+        return data;
+      })
+      .then(data => {
+        hypertopic.post(data);
+        let newState = this.state;
+        newState.topic[viewpointId].push({
+          viewpoint: viewpointId,
+          id: topicToAssign.id
+        });
+        this.setState(newState);
+      })
+      .catch(error => console.log(`error : ${error}`));
+  }
+
+
+  _removeTopic(topicToDelete) {
+    let params = this.props.match.params;
+    let hypertopic = new Hypertopic(conf.services);
+
+    if (confirm('Voulez-vous vraiment retirer le topic de cette oeuvre ?')) {
```

Comment:
Should `react-confirm` (or a similar package) be imported?

Reply:
It doesn't seem to work with `react-confirm`... Or is it just `window.confirm()`?

Reply:
You are right, it's window.confirm() which is tried to be used here.

Reply:
I had to update `react-scripts` and new versions require `window` members to be prefixed. It's OK, I changed the code myself.